google/data_source_google_compute_default_service_account.go

```diff
@@ -0,0 +1,44 @@
+package google
+
+import (
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func dataSourceGoogleComputeDefaultServiceAccount() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceGoogleComputeDefaultServiceAccountRead,
+		Schema: map[string]*schema.Schema{
+			"email": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"project_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+		},
+	}
+}
+
+func dataSourceGoogleComputeDefaultServiceAccountRead(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	project, err := getProject(d, config)
+	if err != nil {
+		return err
+	}
+
+	if v, ok := d.GetOk("project_id"); ok {
```

Comment:
the getProject method checks first if the "project" field is set and then fallback on the default provider project. This means you won't need this check here.